In [10]:
import pandas as pd
dataset=pd.read_csv('general_data_attr_change.csv')
dataset.head()

Age  attr_in_int Attrition     BusinessTravel              Department  \
0   51            0        No      Travel_Rarely                   Sales   
1   31            1       Yes  Travel_Frequently  Research & Development   
2   32            0        No  Travel_Frequently  Research & Development   
3   38            0        No         Non-Travel  Research & Development   
4   32            0        No      Travel_Rarely  Research & Development   

   DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  ...  \
0                 6          2  Life Sciences              1           1  ...   
1                10          1  Life Sciences              1           2  ...   
2                17          4          Other              1           3  ...   
3                 2          5  Life Sciences              1           4  ...   
4                10          1        Medical              1           5  ...   

  NumCompaniesWorked  Over18 PercentSalaryHike StandardHours  \
0                1.0       Y                11             8   
1                0.0       Y                23             8   
2                1.0       Y                15             8   
3                3.0       Y                11             8   
4                4.0       Y                12             8   

   StockOptionLevel  TotalWorkingYears TrainingTimesLastYear  YearsAtCompany  \
0                 0                1.0                     6               1   
1                 1                6.0                     3               5   
2                 3                5.0                     2               5   
3                 3               13.0                     5               8   
4                 2                9.0                     2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 25 columns]

In [11]:
dataset.columns

Index(['Age', 'attr_in_int', 'Attrition', 'BusinessTravel', 'Department',
       'DistanceFromHome', 'Education', 'EducationField', 'EmployeeCount',
       'EmployeeID', 'Gender', 'JobLevel', 'JobRole', 'MaritalStatus',
       'MonthlyIncome', 'NumCompaniesWorked', 'Over18', 'PercentSalaryHike',
       'StandardHours', 'StockOptionLevel', 'TotalWorkingYears',
       'TrainingTimesLastYear', 'YearsAtCompany', 'YearsSinceLastPromotion',
       'YearsWithCurrManager'],
      dtype='object')

In [12]:
from scipy.stats import pearsonr
dataset['TotalWorkingYears']=dataset['TotalWorkingYears'].fillna(11.28)
dataset.columns

Index(['Age', 'attr_in_int', 'Attrition', 'BusinessTravel', 'Department',
       'DistanceFromHome', 'Education', 'EducationField', 'EmployeeCount',
       'EmployeeID', 'Gender', 'JobLevel', 'JobRole', 'MaritalStatus',
       'MonthlyIncome', 'NumCompaniesWorked', 'Over18', 'PercentSalaryHike',
       'StandardHours', 'StockOptionLevel', 'TotalWorkingYears',
       'TrainingTimesLastYear', 'YearsAtCompany', 'YearsSinceLastPromotion',
       'YearsWithCurrManager'],
      dtype='object')

In [13]:
dataset

Age  attr_in_int Attrition     BusinessTravel              Department  \
0      51            0        No      Travel_Rarely                   Sales   
1      31            1       Yes  Travel_Frequently  Research & Development   
2      32            0        No  Travel_Frequently  Research & Development   
3      38            0        No         Non-Travel  Research & Development   
4      32            0        No      Travel_Rarely  Research & Development   
...   ...          ...       ...                ...                     ...   
4405   42            0        No      Travel_Rarely  Research & Development   
4406   29            0        No      Travel_Rarely  Research & Development   
4407   25            0        No      Travel_Rarely  Research & Development   
4408   42            0        No      Travel_Rarely                   Sales   
4409   40            0        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      ... NumCompaniesWorked  Over18 PercentSalaryHike StandardHours  \
0     ...                1.0       Y                11             8   
1     ...                0.0       Y                23             8   
2     ...                1.0       Y                15             8   
3     ...                3.0       Y                11             8   
4     ...                4.0       Y                12             8   
...   ...                ...     ...               ...           ...   
4405  ...                3.0       Y                17             8   
4406  ...                2.0       Y                15             8   
4407  ...                0.0       Y                20             8   
4408  ...                0.0       Y                14             8   
4409  ...                0.0       Y                12             8   

      StockOptionLevel  TotalWorkingYears TrainingTimesLastYear  \
0                    0               1.00                     6   
1                    1               6.00                     3   
2                    3               5.00                     2   
3                    3              13.00                     5   
4                    2               9.00                     2   
...                ...                ...                   ...   
4405                 1              10.00                     5   
4406                 0              10.00                     2   
4407                 0               5.00                     4   
4408                 1              10.00                     2   
4409                 0              11.28                     6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              .

In [14]:
stats, p=pearsonr(dataset.attr_in_int, dataset.Age)
print(stats, p)

-0.15920500686577949 1.996801615887198e-26


As r = -0.159, there’s low negative correlation between Attrition and Age
As the P value of 1.996 is > 0.05, we are accepting H0 and hence there’s no significant correlation
between Attrition & Age


In [15]:
stats, p=pearsonr(dataset.attr_in_int, dataset.DistanceFromHome)
print(stats, p)

-0.009730141010179692 0.5182860428050771


As r = -0.009, there’s low negative correlation between Attrition and DistanceFromHome
As the P value of 0.518 is > 0.05, we are accepting H0 and hence there’s no significant correlation
between Attrition & DistanceFromHome


In [16]:
stats, p=pearsonr(dataset.attr_in_int, dataset.Education)
print(stats, p)

-0.015111167710968721 0.3157293177118575


As r = -0.015, there’s low negative correlation between Attrition and Education
As the P value of 0.315 is > 0.05, we are accepting H0 and hence there’s no significant correlation
between Attrition & Education


In [17]:
stats, p=pearsonr(dataset.attr_in_int, dataset.MonthlyIncome)
print(stats, p)

-0.03117628169811501 0.03842748490600132


As r = -0.031, there’s low negative correlation between Attrition and MonthlyIncome
As the P value of 0.038 is < 0.05, we are accepting Ha and hence there’s significant correlation
between Attrition & MonthlyIncome